microbench/src/main/java/io/netty/microbench/http2/NoPriorityByteDistributionBenchmark.java

```diff
@@ -0,0 +1,315 @@
+/*
+ * Copyright 2015 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License, version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.netty.microbench.http2;
+
+import static org.mockito.Mockito.when;
+
+import io.netty.channel.AbstractChannel;
+import io.netty.channel.ChannelConfig;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelMetadata;
+import io.netty.channel.ChannelOutboundBuffer;
+import io.netty.channel.ChannelPromise;
+import io.netty.channel.DefaultChannelConfig;
+import io.netty.channel.EventLoop;
+import io.netty.handler.codec.http2.DefaultHttp2Connection;
+import io.netty.handler.codec.http2.DefaultHttp2RemoteFlowController;
+import io.netty.handler.codec.http2.Http2Connection;
+import io.netty.handler.codec.http2.Http2Exception;
+import io.netty.handler.codec.http2.Http2RemoteFlowController;
+import io.netty.handler.codec.http2.Http2Stream;
+import io.netty.handler.codec.http2.Http2StreamVisitor;
+import io.netty.handler.codec.http2.PriorityStreamByteDistributor;
+import io.netty.handler.codec.http2.StreamByteDistributor;
+import io.netty.handler.codec.http2.UniformStreamByteDistributor;
+import io.netty.microbench.util.AbstractMicrobenchmark;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.openjdk.jmh.annotations.AuxCounters;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Level;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+
+import java.net.SocketAddress;
+
+/**
+ * Benchmark to compare stream byte distribution algorithms when priorities are identical for
+ * all streams.
+ */
+@Fork(1)
+@State(Scope.Benchmark)
+public class NoPriorityByteDistributionBenchmark extends AbstractMicrobenchmark {
+    public enum Algorithm {
+        PRIORITY,
+        UNIFORM
+    }
+
+    @Param({ "100", "10000" })
+    private int numStreams;
+
+    @Param({ "1024", "65536"/*, "1048576"*/ })
```

Comment:
can you report results with `1048576` after updates are made? I expect them to be good, but better to be sure...